addons/mail/static/src/core/common/message_model.js

```diff
@@ -32,6 +33,24 @@ export class Message extends Record {
     attachment_ids = Record.many("ir.attachment", { inverse: "message" });
     author = Record.one("Persona");
     body = Record.attr("", { html: true });
+    richBody = Record.attr("", {
+        compute() {
+            if (!this.store.emojiLoader.loaded || !this.body) {
+                return this.body;
+            }
+            return htmlReplaceAll(
+                this.body,
+                this.store.emojiLoader.loaded.emojiRegex,
+                (codepoints) =>
+                    markup(
+                        `<span title="${htmlFormatList(
+                            this.store.emojiLoader.loaded.shortcodesByCodepoints[codepoints],
+                            { style: "unit-narrow" }
+                        )}">${htmlEscape(codepoints)}</span>`
+                    )
+            );
+        },
```

Comment:
Note that this is probably a bit too naive. If the Emoji is present in a title (or any other) attribute, we don't want to replace it with a span. We should probably do a more complex lookup, a bit similar as what we do for search highlight probably. Replacement should only ever be done in textContent/text nodes.

Reply:
And even then, we might want to ignore other places, such as pre tags, possibly mentions (if user has emoji in name), code blocks (if/when we have them), ...

Reply:
Logic is extracted into a dedicated function `format.js@wrapEmojisWithTitle`. `WRAP_EMOJI_EXCLUDED_TAGS` constant will allow to easily update the tags we want to exclude. For now, `code`, `pre` and `a` are ignored.